src/model.c

```diff
@@ -192,6 +196,41 @@ void destroy_event_list( model *model, int type )
 	free( model->event_lists[type].events );
 };
 
+void set_up_default_occupation_network( model *model)
+{
+    model->occupation_network = calloc( N_DEFAULT_OCCUPATION_NETWORKS, sizeof( network* ) );
+    for( int idx = 0; idx < N_DEFAULT_OCCUPATION_NETWORKS; idx++ )
+        set_up_occupation_network( model, idx );
+}
+
+void set_up_custom_occupation_network( model *model )
+{
+    long idx, n_people;
+    long *people;
+    double n_interactions;
+    model->occupation_network = calloc( model->n_occupation_networks, sizeof( network* ) );
+
+    for( int network = 0; network < model->n_occupation_networks; network++ )
+    {
+        n_people = 0;
+        people = calloc( model->params->n_total, sizeof( long ) );
+        for( idx = 0; idx < model->params->n_total; idx++ )
+            if( model->population[idx].occupation_network == network )
+                people[n_people++] = idx;
+
+        model->occupation_network[network] = new_network( n_people, OCCUPATION );
+        model->occupation_network[network]->skip_hospitalised = TRUE;
+        model->occupation_network[network]->skip_quarantined  = TRUE;
+        model->occupation_network[network]->daily_fraction    = model->params->daily_fraction_work;
+        model->occupation_network[network]->network_id        = model->params->occupation_network_table->network_ids[network];
+        strcpy( model->occupation_network[network]->name, model->params->occupation_network_table->network_names[network]);
+        n_interactions =  model->params->occupation_network_table->mean_interactions[network] / model->params->daily_fraction_work;
+        build_watts_strogatz_network( model->occupation_network[network], n_people, n_interactions, 0.1, TRUE );
```

Comment:
The hard-coding of 0.1 should be changed to model->params->work_network_rewire. In future versions we may want to have this as part of the occupation table, but let's keep it in line with the main code at the moment.

Also needs the protection of if n_people = 0 (see latest version of `set_up_occupation_network`).

Reply:
Done.  Use `model->params->work_network_rewire` now and added `n_people > 0` check.